### PR TITLE
ui: Remove the e2e-encryption feature from the matrix-sdk-ui crate

### PR DESCRIPTION
**Pull request by poljar**
It does not make much sense to create an UI client that does not support end-to-end encryption, besides disabling the feature was broken for quite some time.

### CODE REVIEW

**Comment by poljar** [comment]
Kicking the CI.

File: crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs [hunk 1]
```diff
@@ -888,11 +888,7 @@ impl<'a, const CAP: usize, Item, Gap> Iterator for IterBackward<'a, CAP, Item, G
     type Item = &'a Chunk<CAP, Item, Gap>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.chunk.map(|chunk| {
-            self.chunk = chunk.previous();
-
-            chunk
-        })
+        self.chunk.inspect(|chunk| self.chunk = chunk.previous())
```
Comment (andybalaam) [inline crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs:891]: Changes in this file seem unrelated?

**Review by andybalaam — CHANGES_REQUESTED** [review]
One extra file seems to have crept in to the PR.

Looks good otherwise, and yay for simplification.

File: crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs:891]: That commit was part of #3958, why didn't github notice :shrug:, it did change the base from that branch to main automagically.

Comment (poljar) [reply crates/matrix-sdk/src/event_cache/linked_chunk/mod.rs]: Rebased locally and force pushed to get rid of the extra commit.

**Review by andybalaam — APPROVED** [review]
Great, thanks!
